rdt/hyper_transformer.py

```diff
@@ -80,73 +78,87 @@ class HyperTransformer:
-            - ``int``: a ``NumericalTransformer`` is created with ``dtype=int``.
-            - ``float``: a ``NumericalTransformer`` is created with ``dtype=float``.
-            - ``object`` or ``category``: a ``CategoricalTransformer`` is created.
-            - ``bool``: a ``BooleanTransformer`` is created.
-            - ``datetime``: a ``DatetimeTransformer`` is created.
-
-        Any other ``dtype`` is not supported and raises a ``ValueError``.
-
-        Args:
-            data (pandas.DataFrame):
-                Data used to analyze the ``pandas.DataFrame`` dtypes.
-
-        Returns:
-            dict:
-                Mapping of column names and transformer instances.
+    @staticmethod
+    def _field_in_data(field, data):
+        all_columns_in_data = isinstance(field, tuple) and all(col in data for col in field)
+        return field in data or all_columns_in_data
 
-        Raises:
-            ValueError:
-                if a ``dtype`` is not supported by the `HyperTransformer``.
-        """
-        transformers = {}
-        if self.dtypes:
-            dtypes = self.dtypes
+    @staticmethod
+    def _add_field_to_set(field, field_set):
+        if isinstance(field, tuple):
+            field_set.update(field)
         else:
-            dtypes = [
-                data[column].dropna().infer_objects().dtype
-                for column in data.columns
-            ]
-
-        for name, dtype in zip(data.columns, dtypes):
-            try:
-                kind = np.dtype(dtype).kind
-            except TypeError:
-                # probably category
-                kind = 'O'
-
-            transformer_template = self.dtype_transformers.get(kind)
-            if not transformer_template:
-                raise ValueError('Unsupported dtype: {}'.format(dtype))
-
-            if isinstance(transformer_template, str):
-                transformer_template = self._TRANSFORMER_TEMPLATES[transformer_template]
-
-            if not isinstance(transformer_template, type):
-                transformer = deepcopy(transformer_template)
+            field_set.add(field)
+
+    def _update_field_types(self, data):
+        # get set of provided fields including multi-column fields
+        provided_fields = set()
+        for field in self.field_types.keys():
+            self._add_field_to_set(field, provided_fields)
+
+        for field in data:
+            if field not in provided_fields:
+                self.field_types[field] = self._DTYPES_TO_DATA_TYPES[data[field].dtype.kind]
+
+    def _fit_field_transformer(self, data, field, transformer):
+        transformer = load_transformer(transformer)
+        transformer.fit(data, field)
+        self._transformers_sequence.append(transformer)
+        output_types = transformer.get_output_types()
+        next_transformers = transformer.get_next_transformers()
+        for (output, output_type) in output_types.items():
+            if output in self.field_transformers:
```

Comment:
I think we could extract this if/else block into a `_get_next_transformer` method, so the logic in the loop becomes shorter:

```python3
for (output, output_type) in output_types.items():
    next_transformer = self._get_next_transformer(output, output_type, data)
    if next_transformer:
        self._temp_columns.append(output)
        transformed_data = transformer.transform(data)
        self._fit_field_transformer(transformed_data, output, next_transformer)
    else:
        self._output_columns.append(output)
```